packages/vscode-extension/src/debugging/DebugAdapter.ts

```diff
@@ -407,12 +434,40 @@ export class DebugAdapter extends DebugSession {
     args: DebugProtocol.VariablesArguments
   ): Promise<void> {
     response.body = response.body || {};
-    response.body.variables = await this.variableStore.get(
-      args.variablesReference,
-      (params: object) => {
-        return this.cdpSession.sendCDPMessage("Runtime.getProperties", params);
+    response.body.variables = [];
+
+    if (args.filter === "indexed") {
+      const stringified = "" + getArraySlots;
+
+      try {
+        const partialValue = await this.cdpSession.sendCDPMessage("Runtime.callFunctionOn", {
+          functionDeclaration: stringified,
+          objectId: this.variableStore.convertDAPObjectIdToCDP(args.variablesReference),
+          arguments: [args.start, args.count].map((value) => ({ value })),
+        });
+
+        const properties = await this.variableStore.get(
+          this.variableStore.adaptCDPObjectId(partialValue.result.objectId),
+          (params: object) => {
+            return this.cdpSession.sendCDPMessage("Runtime.getProperties", params);
+          }
+        );
+
+        response.body.variables = properties;
+      } catch (e) {
+        Logger.error("[CDP] Failed to retrieve array partially", e);
       }
-    );
+    }
+
+    if (args.filter !== "indexed" && args.filter !== "named") {
```

Comment:
maybe this should be an `else if` given we exclude "indexed" here. Also what happens when "named" is used as this doesn't seem to be handled at all?